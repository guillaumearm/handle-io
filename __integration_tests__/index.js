const testHandleIo = (handleIo) => {
  test('should expose handle-io api', () => {
    const { io, handler, testHandler } = handleIo;
    expect(typeof io).toBe('function');
    expect(typeof handler).toBe('function');
    expect(typeof testHandler).toBe('function');
    expect(Object.keys(handleIo)).toEqual(['io', 'handler', 'testHandler']);
  });
}

describe('integration', () => {
  describe('commonjs', () => {
    // eslint-disable-next-line node/no-missing-require
    testHandleIo(require('../lib/handle-io.js'));
  });

  describe('es module', () => {
    // eslint-disable-next-line node/no-missing-require
    testHandleIo(require('../es/handle-io.js'));
  });

  describe('umd module', () => {
    // eslint-disable-next-line node/no-missing-require
    testHandleIo(require('../dist/handle-io.js'));
  });
});
