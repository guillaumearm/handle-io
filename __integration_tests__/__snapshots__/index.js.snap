// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration commonjs should expose valid handle-io api 1`] = `
Object {
  "handler": [Function],
  "io": [Function],
  "testHandler": [Function],
}
`;

exports[`integration es module should expose valid handle-io api 1`] = `
Object {
  "handler": [Function],
  "io": [Function],
  "testHandler": [Function],
}
`;

exports[`integration umd module should expose valid handle-io api 1`] = `
Object {
  "handler": [Function],
  "io": [Function],
  "testHandler": [Function],
}
`;
